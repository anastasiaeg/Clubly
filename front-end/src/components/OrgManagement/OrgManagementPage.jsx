import React from "react";
import "./orgmanagement.css";
import { Link } from "react-router-dom";
import * as routes from "../../constants/routes";

export default class OrgManagementPage extends React.Component {
  render() {
    const { user } = this.props;

    return (
      <div className="event-management">
        <h1>Managed organizations</h1>
        <div className="image-gallery">
          {user &&
            user.organizerOf.map(club => (
              <div className="gallery-item" key={club.id}>
                <Link to={routes.ORG + "/" + encodeURI(club.id)}>
                  <img src={club.image} alt={club.name} />
                </Link>
                <div className="description">{club.name}</div>
              </div>
            ))}
        </div>
      </div>
    );
  }
}
