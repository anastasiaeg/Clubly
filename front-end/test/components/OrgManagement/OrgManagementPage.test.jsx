import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import OrgManagementPage from '../../../src/components/OrgManagement/OrgManagementPage';

describe('Org Management Page', () => {
    test('should render', () => {
        const component = renderer.create(
            <MemoryRouter>
                <OrgManagementPage/>
            </MemoryRouter>
        )
        .toJSON();
        expect(component).toMatchSnapshot();
    });
})