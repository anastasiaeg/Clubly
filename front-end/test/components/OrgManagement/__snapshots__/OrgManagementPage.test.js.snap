// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Org Management Page should render 1`] = `
<div
  className="event-management"
>
  <h1>
    Managed organizations
  </h1>
  <div
    className="image-gallery"
  >
    <div
      className="gallery-item"
    >
      <a
        href="/org/undefined"
        onClick={[Function]}
      >
        <img
          alt={undefined}
          src={undefined}
        />
      </a>
      <div
        className="description"
      />
    </div>
  </div>
</div>
`;
